## Задача 1
- Скачать файл в уроке
- Считать данные с помощью pandas
- Вывести на экран первые 5 строк
- Посмотреть на описание признаков и на их содержание

In [2]:
import pandas as pd

df = pd.read_csv('kc_house_data.csv', sep = ',')
df.head()


id             date     price  bedrooms  bathrooms  sqft_living   
0  7129300520  20141013T000000  221900.0         3       1.00         1180  \
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement   
0      5650     1.0           0     0  ...      7        1180              0  \
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
0      1955             0    98178  47.5112 -122.257           1340  \
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

## Задача 2
- Проведите первичный анализ данных
- Изучите типы данных
- Найдите количество пропущенных ячеек в данных
- Посчитайте основные статистики по всем признакам и поизучайте их
- Пишите выводы

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

#### Мои выводы:
- в таблице 21 столбец и 21613 строк с данными
- типы данных: строки, целые и вещественные числа
- нет пустых ячеек


##  Задача 3

### Задача 3.1

В каком диапазоне изменяются стоимости недвижимости?


In [4]:
df['price'].min(), df['price'].max()

(75000.0, 7700000.0)

### Задача 3.2

Какую долю в среднем занимают жилая площадь от всей площади по всем домам?


In [6]:
sqrt_liv = df['sqft_living'].mean()
sqrt_lot = df['sqft_lot'].mean()
sqrt_liv / sqrt_lot * 100

13.76781757959227

### Задача 3.3

Как много домов с разными этажами в данных?


In [40]:
df['floors'].value_counts()

floors
1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: count, dtype: int64

### Задача 3.4

Насколько хорошие состояния у домов в данных?


In [41]:

df['condition'].value_counts()

condition
3    14031
4     5679
5     1701
2      172
1       30
Name: count, dtype: int64

### Задача 3.5

Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [23]:
df['yr_built'].min(), df['yr_built'].max()


(1900, 2015)

## Задача 4

Ответьте на несколько вопросов


### Задача 4.1

Сколько в среднем стоят дома, у которых 2 спальни?


In [24]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

### Задача 4.2

 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [25]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

### Задача 4.3

Как много домов коснулся ремонт?


In [42]:
df[df['yr_renovated'] != 0]['id'].shape[0]

914

### Задача 4.4

Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [29]:
bad = df[df['grade'] < 4]['price'].mean()
good = df[df['grade'] > 10]['price'].mean()
good - bad

1488885.1175298805

## Задача 5

Ответьте на несколько вопросов

### Задача 5.1

Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [44]:
df[(df['waterfront'] == 1) & (df['bathrooms'] > 3) & (df['sqft_basement'] != 0)].shape[0]

41

### Задача 5.2
Выберите дом клиенту

Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [48]:
df[(df['condition'] == 5) & (df['yr_built'] >= 1980) & ((df['waterfront'] == 1) | (df['view'] == 4))]['price'].min(), df[(df['condition'] == 5) & (df['yr_built'] >= 1980) & ((df['waterfront'] == 1) | (df['view'] == 4))]['price'].max()

(1295000.0, 3000000.0)

### Задача 5.3
Выберите дом клиенту

Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [39]:
df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]['condition'].mean()

2.8333333333333335